ethrex/src/main.rs

```diff
@@ -25,17 +32,68 @@ fn main() {
     let authrpc_port = matches
         .get_one::<String>("authrpc.port")
         .expect("authrpc.port is required");
+
+    let tcp_addr = matches.get_one::<String>("addr").expect("addr is required");
+    let tcp_port = matches.get_one::<String>("port").expect("port is required");
+    let udp_addr = matches
+        .get_one::<String>("discovery.addr")
+        .expect("discovery.addr is required");
+    let udp_port = matches
+        .get_one::<String>("discovery.port")
+        .expect("discovery.port is required");
+
     let genesis_file_path = matches
         .get_one::<String>("network")
         .expect("network is required");
 
+    let bootnode_string = matches
+        .get_one::<String>("bootnodes")
+        .expect("bootnode is required");
+    let bootnode = parse_bootnode(&bootnode_string);
+
+    let http_socket_addr =
+        parse_socket_addr(http_addr, http_port).expect("Failed to parse http address and port");
+    let authrpc_socket_addr = parse_socket_addr(authrpc_addr, authrpc_port)
+        .expect("Failed to parse authrpc address and port");
+
+    let udp_socket_addr =
+        parse_socket_addr(udp_addr, udp_port).expect("Failed to parse discovery address and port");
+    let tcp_socket_addr =
+        parse_socket_addr(tcp_addr, tcp_port).expect("Failed to parse addr and port");
+
     let _genesis = read_genesis_file(genesis_file_path);
 
-    rpc::start_api(http_addr, http_port, authrpc_addr, authrpc_port);
+    let rpc_api = rpc::start_api(http_socket_addr, authrpc_socket_addr);
+    let networking = net::start_network(udp_socket_addr, tcp_socket_addr);
+
+    join!(rpc_api, networking);
 }
 
-fn read_genesis_file(genesis_file_path: &String) -> Genesis {
+fn read_genesis_file(genesis_file_path: &str) -> Genesis {
     let genesis_file = std::fs::File::open(genesis_file_path).expect("Failed to open genesis file");
     let genesis_reader = BufReader::new(genesis_file);
     serde_json::from_reader(genesis_reader).expect("Failed to read genesis file")
 }
+
+fn parse_socket_addr(addr: &str, port: &str) -> Result<SocketAddr, AddrParseError> {
+    format!("{addr}:{port}").parse()
+}
+
+fn parse_bootnode(input: &str) -> BootNode {
```

Comment:
should we add a From implementation inside BootNode and move this logic there?

Reply:
Definitely. 
What do you think about storing the `node_id` in a `Vec<u8>`? Should I keep it that way or change it to a `U512`? or maybe to another type?

Reply:
U512 sounds like a good option

Reply:
Let's use `H512` since it's a hash